### PR TITLE
[v13] Bump tough-cookie from 4.1.2 to 4.1.3

### PR DESCRIPTION
**Pull request by jentfoo**
Our `sec-scan` repo's discovered that `tough-cookie` was not fully updated on v13 for `CVE-2023-26136`: https://github.com/gravitational/teleport-sec-scan-3/security/dependabot/157

Initial update attempt: https://github.com/gravitational/teleport/pull/28914

Bumps [tough-cookie](https://github.com/salesforce/tough-cookie) from 4.1.2 to 4.1.3.
- [Release notes](https://github.com/salesforce/tough-cookie/releases)
- [Changelog](https://github.com/salesforce/tough-cookie/blob/master/CHANGELOG.md)
- [Commits](https://github.com/salesforce/tough-cookie/compare/v4.1.2...v4.1.3)

### CODE REVIEW

**Comment by ravicious** [comment]
There's something weird going on with `@types/tough-cookie`. If you look at yarn.lock in branch/v13, `@types/tough-cookie` is resolved to the `tough-cookie` package. [I already ran into this in the past](https://github.com/gravitational/teleport/pull/33742#issuecomment-1772769061), but only now I realized what's wrong. I don't know how this happened, maybe it was a conflict resolution gone wrong.

In any case, I made another branch (https://github.com/gravitational/teleport/pull/36792) where I've fixed the problem with `@types/tough-cookie` and updated `tough-cookie`.

I've done this by:

1. Removing from package.json all packages that depend on `@types/tough-cookie` (`@types/jsdom`, `jest-environment-jsdom`) and running yarn so that `@types/tough-cookie` gets removed from yarn.lock.
2. Adding those packages back again and running yarn so that `@types/tough-cookie` gets resolved properly.
3. Adding `"tough-cookie": "^4.1.3"` to web/packages/build/package.json so that yarn updates tough-cookie.
4. Removing tough-cookie from that package.json since it's actually only a transitive dependency.

Your PR has a good description already. Could you hard reset your branch to point at the tip of mine so that we merge the tough-cookie update with all the yarn problems resolved?

`@types/tough-cookie` appears to be resolved correctly on branch/v15 and branch/v14.

**Review by ravicious — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -3728,14 +3728,9 @@
...
 "@types/tough-cookie@*":
-  version "4.1.3"
-  resolved "https://registry.yarnpkg.com/tough-cookie/-/tough-cookie-4.1.3.tgz#97b9adb0728b42280aa3d814b6b999b2ff0318bf"
-  integrity sha512-aX/y5pVRkfRnfmuX+OdbSdXvPe6ieKX/G2s7e98f4poJHnqH3281gDPm/metm6E/WRamfx7WC4HUqkWHfQHprw==
-  dependencies:
-    psl "^1.1.33"
-    punycode "^2.1.1"
-    universalify "^0.2.0"
-    url-parse "^1.5.3"
+  version "4.0.5"
```
Comment (codingllama) [inline yarn.lock:3731]: Is this correct? Looks like a downgrade.

**Review by codingllama — APPROVED** [review]
bot

File: yarn.lock (hunk 1 above)
Comment (jentfoo) [reply yarn.lock:3731]: I raised this as well: https://github.com/gravitational/teleport/pull/36792#discussion_r1455887514

The prior version was seemingly in error, once we merge I will review what Dependabot says

**Pull request merged**